sound/soc/intel/boards/sof_sdw.c

```diff
@@ -1175,6 +1175,15 @@ static int create_codec_dai_name(struct device *dev,
 
 		adr = link->adr_d[i].adr;
 
+		codec_index = find_codec_info_part(adr);
+		if (codec_index < 0)
+			return codec_index;
+		if (_codec_index != -1 && codec_index != _codec_index) {
+			dev_dbg(dev, "Different devices on the same sdw link\n");
+			break;
+		}
+		_codec_index = codec_index;
+
```

Comment:
I find the commit message a bit cryptic @bardliao, does this fix a problem with a board or specific configuration? Or is this more of a defensive programming protection to avoid doing bad things?

Reply:
> I find the commit message a bit cryptic @bardliao, does this fix a problem with a board or specific configuration? Or is this more of a defensive programming protection to avoid doing bad things?

@plbossart It is more of a defensive programming protection to avoid doing bad things. Currently, we won't meet any issue because there is no different codec in a sdw link so far. However, once we have different codecs in a sdw link, the `codec[comp_index].name` will be set to an incorrect string. But in theory, the `codec[comp_index].name` will be set to the correct string when `create_codec_dai_name` is called for the second codec in the sdw link.